apollo-ios/Sources/Apollo/FieldSelectionCollector.swift

```diff
@@ -93,15 +93,26 @@ struct DefaultFieldSelectionCollector: FieldSelectionCollector {
                             info: info)
         }
 
-      case let .deferred(_, typeCase, _):
+      case let .deferred(condition, typeCase, _):
         // In Apollo's implementation (Router + Server) of deferSpec=20220824 ALL defer directives
         // will be honoured and sent as separate incremental responses. This means deferred
         // selection fields never need to be collected because they are parsed with the incremental
         // data, at which time they are no longer deferred.
         //
         // The deferred fragment identifiers still need to be collected becuase that is how the
         // user determines the state of the deferred fragment via the @Deferred property wrapper.
-        groupedFields.addDeferredFragment(typeCase)
+        var collectDeferredFragment: Bool = true
```

Comment:
I think we should change this to be named `isDeferred` with a comment saying, if the condition resolves to false, the fragment is collected normally and not deferred.